### PR TITLE
Kerbos services and more ssl protocol support

### PR DESCRIPTION
**Pull request by axeisghost**

### CODE REVIEW

**Comment by joshwalters** [comment]
When you make the pull request, remember to give credit to the Impyla source (https://github.com/cloudera/impyla/blob/255b07ed973d47a3395214ed92d35ec0615ebf62/impala/_thrift_api.py#L152-L160). It is also apache 2.

File: pyhive/hive.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def connect(*args, **kwargs):
 class Connection(object):
     """Wraps a Thrift session"""
 
-    def __init__(self, host, port=10000, username=None, database='default', auth='NONE',
+    def __init__(self, host, kerberos_service_name, port=10000, username=None, password=None, database='default', auth='NONE',
```
Comment (joshwalters) [inline pyhive/hive.py]: The `kerberos_service_name` param, what if there is not one provided? Not all users will be running kerberos. Also, new api additions should go at the end of the method, and maybe they should have `...=None` default values?

**Comment by joshwalters** [comment]
This looks good to me. Have you tested and verified that it works as well?

**Comment by axeisghost** [comment]
I tested it with jetblue, it works well of course but I havent tested other mechanism since I do not know or access any other Hive clusters that is using different mechanism

**Pull request closed without merging**